packages/ngx-flicking/projects/ngx-flicking/src/lib/ngx-flicking.component.ts

```diff
@@ -133,8 +183,38 @@ export class NgxFlickingComponent implements OnInit, AfterViewInit, OnDestroy, O
 
         if (emitter) {
           emitter.emit(e);
+
+          if (eventName === 'visibleChange') {
+            const list = counter(this.panels.length * (this.flicking.getCloneCount() + 1));
+            const min = e.range.min;
+            const max = e.range.max;
+
+            const visibles = min >= 0
+              ? list.slice(min, max + 1)
+              : list.slice(0, max + 1).concat(list.slice(min));
+
+            this.triggerRenderChange(visibles);
+          }
         }
       });
     });
   }
+
+  private triggerRenderChange(visibles: number[]) {
+    // visible panel is not changed
+    if (visibles.length === this.prevVisibles.length &&
+        visibles.every((v, i) => this.prevVisibles[i] === v)) {
+      return;
+    }
+
+    this.prevVisibles = visibles;
+
+    if (this.criticalSection) {
+      // Use microtask to run as soon as synchronous job is done.
+      Promise.resolve()
+        .then(() => this.renderPanelChange.emit(visibles));
+    } else {
+      this.renderPanelChange.emit(visibles);
```

Comment:
@jongmoon 
What about sending events as objects rather than as arrays?

```js
emit({
    visibles,
});

Reply:
@daybrush I try to change it. check it please after I pull request.